src/main/java/gregtech/common/covers/CoverRoboticArm.java

```diff
@@ -115,24 +121,24 @@ protected int doTransferExact(IItemHandler itemHandler, IItemHandler myItemHandl
     }
 
     protected int doKeepExact(IItemHandler itemHandler, IItemHandler myItemHandler, int maxTransferAmount) {
-        Map<Object, GroupItemInfo> currentItemAmount = doCountDestinationInventoryItemsByMatchIndex(itemHandler,
+        Map<Integer, GroupItemInfo> currentItemAmount = doCountDestinationInventoryItemsByMatchIndex(itemHandler,
                 myItemHandler);
-        Map<Object, GroupItemInfo> sourceItemAmounts = doCountDestinationInventoryItemsByMatchIndex(myItemHandler,
+        Map<Integer, GroupItemInfo> sourceItemAmounts = doCountDestinationInventoryItemsByMatchIndex(myItemHandler,
                 itemHandler);
-        Iterator<Object> iterator = sourceItemAmounts.keySet().iterator();
+        Iterator<Integer> iterator = sourceItemAmounts.keySet().iterator();
         while (iterator.hasNext()) {
-            Object filterSlotIndex = iterator.next();
+            int filterSlotIndex = iterator.next();
```

Comment:
There is risk of an NPE here if the iterator permits null values. This should be double checked for safety before merging.

Reply:
the iterator comes from an `Int2ObjectOpenHashMap`, whose keys are ints and shouldn't be null